gdal/frmts/netcdf/netcdflayer.cpp

```diff
@@ -2341,9 +2547,18 @@ OGRErr netCDFLayer::CreateField(OGRFieldDefn *poFieldDefn, int /* bApproxOK */)
             {
                 if( m_nDefaultMaxWidthDimId < 0 )
                 {
-                    status =
-                        nc_def_dim(m_nLayerCDFId, "string_default_max_width",
-                                   m_nDefaultWidth, &m_nDefaultMaxWidthDimId);
+                    if(!m_bLegacyCreateMode)
```

Comment:
see above remarks. could be merged in a single code path with a passthrough vcdf implementation